app/controllers/api/v1/admin/admin_api_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Api
+    module V1
+        module Admin
+            class AdminApiController < Api::V1::ApiController
+                before_action :authenticate_admin!
+            
+                def authenticate_admin!
+                    if current_api_v1_user && current_api_v1_user['is_active'] && current_api_v1_user['is_admin'] 
+                        true
+                    else
+                        render json: { error: 'Unauthorized' }, status: :unauthorized
```

Comment:
This is ok, but a better way would be raising an exception and catching it in the exception handler